### ISSUE TITLE
runtime error after i/o timeout

### ISSUE THREAD

**Issue by scross01** [scross01/prometheus-klipper-exporter#31]
> Hello again, I am having the same issue when the node is off, triggering an i/o timeout and eventually a nil pointer dereference.

```
Jul 09 23:55:05 yix prometheus-klipper-exporter[194451]: time="2024-07-09T23:55:05+03:00" level=error msg="Get \"http://my-printer:7125/server/history/totals\": dial tcp [fd7a:115c:a1e0:ab12:4843:cd96:6262:1a38]:7125: i/o timeout"
Jul 09 23:55:05 yix prometheus-klipper-exporter[194451]: panic: runtime error: invalid memory address or nil pointer dereference
```

_Originally posted by @ctrochalakis in https://github.com/scross01/prometheus-klipper-exporter/issues/24#issuecomment-2219743620_

**Comment by scross01** [comment]
I think I see what the problem could be, if the klipper host stops responding part way through the collection it reports the error, but then carries on assuming it has the required metric data to populate the collector.  An updated is needed to catch the exceptions correctly mid collection.

**Comment by scross01** [comment]
Issue was resolved in v0.11.1
